frontend/src/components/ExcelUpload.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+import { ExcelToSchedule } from "../utils/excelParser"
+import { Schedule } from "../models/types"
+import styled from "styled-components";
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+
+
+const Container = styled.div`
+    position: relative;
+    justify-content: right;
+    padding: 0px;
+    margin: 0px;
+`;
+
+
+interface props {
+    setSchedule: ((schedule: Schedule) => void)
+}
+
+export class ExcelUpload extends React.Component<props> {
+    constructor(props: any) {
+        super(props);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    async handleChange(e : any) {
+        if (e != null && e.target != null) {
+            console.log(e.target.files);
+            const file = e.target.files[0]
+            ExcelToSchedule(file, this.props.setSchedule)
```

Comment:
Is it possible to make this an async function and have it return a schedule, instead of having to pass in a callback?

Reply:
Talked on slack and decided to keep the callback